lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -473,6 +495,7 @@ def __init__(
         port: Optional[int] = None,
         strip_prefix: bool = False,
         redirect_https: bool = False,
+        scheme: Union[Scheme, SchemeLiteral] = Scheme.HTTP,
```

Comment:
```suggestion
        scheme: SchemeLiteral = "http",
```

Reply:
Could you open a tandem PR where this lib is used? (alertmanager is IPA)

In our existing pattern, we may not know in charm's `__init__` if it's going to be http or https, because we will be waiting for a cert on a different relation.

Need to make sure this works as expected in practice.

Reply:
I wonder what happens when traefik has `http` in its `lb` list, but the endpoint is behind TLS. Would traefik switch (and remember) to use https on its own? Some clients may return with code `497 HTTP Request Sent to HTTPS Port` ("Client sent an HTTP request to an HTTPS server"). Maybe traefik picks it up and switches?

Reply:
tandem pr: https://github.com/canonical/alertmanager-k8s-operator/pull/172